src/pages/ExpiringResults/index.js

```diff
@@ -0,0 +1,295 @@
+import React, { Component } from 'react';
+import { connect } from 'dva';
+import {
+  Grid,
+  GridItem,
+  Card,
+  TextContent,
+  Text,
+  TextVariants,
+  Button,
+  Progress,
+  ProgressSize,
+  ProgressMeasureLocation,
+  ProgressVariant,
+  Tooltip,
+} from '@patternfly/react-core';
+import { Icon } from 'antd';
+import { EllipsisVIcon } from '@patternfly/react-icons';
+import { formatDate, getDiffDays, getDiffDate } from '../../utils/moment_constants';
+import Table from '@/components/Table';
+import styles from './index.less';
+
+const expirationLimit = { val: 30 };
+
+@connect(({ global, user, dashboard }) => ({
+  selectedDateRange: global.selectedDateRange,
+  selectedControllers: global.selectedControllers,
+  user: user.user,
+  seenResults: user.seenResults,
+  favoriteControllers: user.favoriteControllers,
+  results: dashboard.results,
+}))
+class ExpiringResults extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      totalResultData: [],
+    };
+  }
+
+  componentDidMount() {
+    const { dispatch, selectedControllers, selectedDateRange } = this.props;
+    dispatch({
+      type: 'dashboard/fetchResults',
+      payload: {
+        selectedDateRange,
+        controller: selectedControllers,
+      },
+    }).then(() => {
+      const { results } = this.props;
+      console.log(results[selectedControllers[0]]);
+      this.setState({
+        totalResultData: results[selectedControllers[0]],
+      });
+    });
+  }
+
+  favoriteRecord = (event, value, controller) => {
+    // Stop propagation from going to the next page
+    event.stopPropagation();
+    const { dispatch } = this.props;
+    // dispatch an action to favorite controller
+    dispatch({
+      type: 'user/favoriteController',
+      payload: controller,
+    });
+  };
+
+  removeControllerFromFavorites = (event, value, controller) => {
+    // Stop propagation from going to the next page
+    event.stopPropagation();
+    const { dispatch } = this.props;
+    // dispatch an action to favorite controller
+    dispatch({
+      type: 'user/removeControllerFromFavorites',
+      payload: controller,
+    });
+  };
+
+  deleteResult = rows => {
+    const keys = rows.map(({ original }) => original.key);
+    const { totalResultData } = this.state;
+    const { dispatch } = this.props;
+    const updatedResult = totalResultData.filter(item => !keys.includes(item.key));
+    dispatch({
+      type: 'dashboard/updateResults',
+      payload: updatedResult,
+    });
+  };
+
+  removeResultFromSeen = controller => {
+    console.log(controller);
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'user/removeResultFromSeen',
+      payload: controller,
+    });
+  };
+
+  showDrowpdown = id => {
+    const dropdownElem = document.getElementById(id);
```

Comment:
We should avoid using the DOM API and use PatternFly's exposed API for displaying and hiding the dropdown menu.

Reply:
Since we aren't using PatternFly's table, we can't replicate the actions column used in PF tables. Since the dropdowns are created dynamically, it's difficult to use `refs` to correctly address them. @aquibbaig any suggestions on how to fix this issue without direct dom manipulation